src/adapters/discord.js

```diff
@@ -1,142 +1,112 @@
 const Discord = require('discord.js');
 const debug = require('debug')('bot:adapter:discord');
 
-const isString = require('lodash/isString');
-const isObject = require('lodash/isObject');
-const isArray = require('lodash/isArray');
-const isEmpty = require('lodash/isEmpty');
+const merge = require('lodash/merge');
+const pick = require('lodash/pick');
 
 // const { PREFIX } = require('./../config');
 const { discord: discordCfg } = require('../config');
 
 const discordAdapter = () => {};
 
-discordAdapter.__INIT__ = function ({ process }) {
+discordAdapter.__INIT__ = function (ctx) {
     const discordBot = new Discord.Client();
 
     discordBot.on('ready', () => {
         debug('Discrodjs ready');
     });
 
-    discordBot.on('message', (msg) => {
+    const handler = async (output) => {
+        const { message, to, reactions } = output;
+        const [, channelId, msgId] = to;
```

Comment:
unclear first elem

Reply:
adapterName, we can skip it, no need decompose